logstash-core/lib/logstash/patches/clamp.rb

```diff
@@ -0,0 +1,69 @@
+require 'clamp'
+require 'logstash/environment'
+
+module Clamp
+  module Attribute
+    class Instance
+      def default_from_environment
```

Comment:
from what I understand in https://github.com/mdub/clamp/blob/master/lib/clamp/option/parsing.rb#L39-L42, clamp always tries to fetch from environment, I believe :environment_variable is just available if you want to use a different name from the inferred one.